yarn-project/pxe/src/synchronizer/synchronizer.ts

```diff
@@ -320,4 +334,67 @@ export class Synchronizer {
       notes: Object.fromEntries(this.noteProcessors.map(n => [n.publicKey.toString(), n.status.syncedToBlock])),
     };
   }
+
+  /**
+   * Retry decoding any deferred notes for the specified contract address.
+   * @param contractAddress - the contract address that has just been added
+   */
+  public async reprocessDeferredNotesForContract(contractAddress: AztecAddress) {
```

Comment:
What do you think about running this function in the `jobQueue` so that it has a consistent view of nullifiers and db state :thinking:.